pages/_app.tsx

```diff
@@ -3,12 +3,36 @@ import { ThemeProvider } from '@material-ui/core/styles'
 import App from 'next/app'
 import Head from 'next/head'
 import React from 'react'
+import { blue, deepPurple } from '@material-ui/core/colors'
+import { Color } from '@material-ui/core'
 
-import theme from '../components/theme'
+import { MakeTheme } from '../components/Theme'
+import { AppContext } from '../components/AppContext'
+
+interface State {
+	themeType: 'dark' | 'light'
+	primary: Color
+	secondary: Color
+}
 
 // `App` component is used for initialize of pages. It allows for overwriting and full control of the `page` initialization.
 // This allows for keeping state between navigation, custom error handling, injecting additional data.
 export default class MyApp extends App {
+	state: State = {
+		themeType: 'dark',
+		primary: blue,
+		secondary: deepPurple,
+	}
+
+	setThemeType = (type: string): void => {
+		if (type === 'dark' || type === 'light') {
```

Comment:
Using `PaletteType` instead of `string` would allow for static type-checking, so no need to check `type === 'dark' || type === 'light'` at runtime.

Might also be worth a comment saying why `setThemeType` has to be an arrow func - i.e. `this` gets mis-bound when calling via `NavBar`'s `useContext(AppContext)`.